cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -281,92 +286,75 @@ mkTracers traceOptions tracer = do
             Consensus.TraceForgedInvalidBlock slot _ _ ->
               LogValue "forgedInvalidSlotLast" $ PureI $ fromIntegral $ unSlotNo slot
 
-    enableConsensusTracer
-      :: Show a
-      => (ConsensusTraceOptions -> Const Bool b)
-      -> Tracer IO String -> Tracer IO a
-    enableConsensusTracer f = if getConst $ f $ traceConsensus traceOptions
-      then showTracing
-      else const nullTracer
-    hasConsensusTraceFlag
-      :: (ConsensusTraceOptions -> Const Bool b)
-      -> Bool
-    hasConsensusTraceFlag f = getConst $ f $ traceConsensus traceOptions
-
-    mkConsensusTracers :: ForgeTracers -> Consensus.Tracers' peer blk (Tracer IO)
-    mkConsensusTracers forgeTracers = Consensus.Tracers
+    mkConsensusTracers :: ForgeTracers -> TraceOptions -> Consensus.Tracers' peer blk (Tracer IO)
+    mkConsensusTracers forgeTracers traceOpts = Consensus.Tracers
       { Consensus.chainSyncClientTracer
-        = annotateSeverity $ filterSeverity (pure . const (tracingSeverity $ hasConsensusTraceFlag Consensus.chainSyncClientTracer))
-          $ toLogObject' (tracingFormatting $ hasConsensusTraceFlag Consensus.chainSyncClientTracer) tracingVerbosity
+        = tracerOnOff (traceChainSyncClient traceOpts)
+          $ toLogObject' StructuredLogging tracingVerbosity
           $ addName "ChainSyncClient" tracer
       , Consensus.chainSyncServerHeaderTracer
-        = annotateSeverity $ filterSeverity (pure . const (tracingSeverity $ hasConsensusTraceFlag Consensus.chainSyncServerHeaderTracer))
-          $ toLogObject' (tracingFormatting $ hasConsensusTraceFlag Consensus.chainSyncServerHeaderTracer) tracingVerbosity
+        =  tracerOnOff (traceChainSyncHeaderServer traceOpts)
+          $ toLogObject' StructuredLogging tracingVerbosity
           $ addName "ChainSyncHeaderServer" tracer
       , Consensus.chainSyncServerBlockTracer
-        = annotateSeverity $ filterSeverity (pure . const (tracingSeverity $ hasConsensusTraceFlag Consensus.chainSyncServerBlockTracer))
-          $ toLogObject' (tracingFormatting $ hasConsensusTraceFlag Consensus.chainSyncServerBlockTracer) tracingVerbosity
+        = tracerOnOff (traceChainSyncBlockServer traceOpts)
+          $ toLogObject' StructuredLogging tracingVerbosity
           $ addName "ChainSyncBlockServer" tracer
       , Consensus.blockFetchDecisionTracer
-        = annotateSeverity $ filterSeverity (pure . const (tracingSeverity $ hasConsensusTraceFlag Consensus.blockFetchDecisionTracer))
-          $ teeTraceBlockFetchDecision (tracingFormatting $ hasConsensusTraceFlag Consensus.blockFetchDecisionTracer) tracingVerbosity
+        = tracerOnOff (traceBlockFetchDecisions traceOpts)
+          $ annotateSeverity $ filterSeverity (pure . const (tracingSeverity True))
```

Comment:
`filterSeverity` can be removed everywhere since the tracer is either on or off.